File: 198-house-robber/house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n=nums.size();
        
        int prev=nums[0],prev2=0;int pick,npick;
        for(int i=1;i<n;++i)
        {   if(i>1)
                pick=nums[i]+prev2;

            else pick=nums[i];
                npick=0+prev;
            int curi=max(pick,npick);
            prev2=prev;
            prev=curi;
        }
        return prev;
    }
};

File: 198-house-robber/house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n=len(nums)
        dp=[0]*(n+1)
        dp[0]=0
        dp[1]=nums[0]
        

        dp[1]=nums[0]
        for i in range(1,n):
            dp[i+1]=max(dp[i-1]+nums[i],dp[i])
        return dp[n]